File: 1055-pairs-of-songs-with-total-durations-divisible-by-60/1055-pairs-of-songs-with-total-durations-divisible-by-60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        int n = time.size();
        map<int,int>mp;
        int ans = 0;

        for(int i=0; i<n; ++i) {
            ans += (mp[(60 - time[i] % 60) % 60]);
            mp[time[i]%60]++;
        }
        return ans;
    }
};

File: 1055-pairs-of-songs-with-total-durations-divisible-by-60/1055-pairs-of-songs-with-total-durations-divisible-by-60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int n = time.length;

        Map<Integer, Integer>mp = new HashMap<>();

        int ans = 0;
        for(int i=0; i<n; ++i) {
            int x = (60 - time[i] % 60) % 60;
            ans += mp.getOrDefault(x, 0);
            int y = time[i] % 60;

            mp.put(y, mp.getOrDefault(y, 0) + 1);
        }
        return ans;
    }
}
